File: src/search/03binary-search.ts
function binarySearch<A>(list: Array<A>, item: A): number | null {

    let left:  number = 0
    let right: number = list.length

    let point: number

    while (true) {

        point = Math.floor( (left + right) / 2 )

        if (list[point] === item) {
            return point
        }
        if (left + 1 === right) {
            return null
        }

        if (item < list[point]) {
            right = point
        }
        else {
            left = point
        }
    }
}

export default binarySearch

File: src/search/03binary-search.py
import data as data



def binarySearch(list, item):

    left  = 0
    right = len(list)

    point = None

    while (True):

        point = round((left + right) / 2)

        if (list[point] == item):
            return point
        if (left + 1 >= right):
            return None



        if (item < list[point]):
            right = point
        else:
            left = point

print(
    binarySearch(data.A, 'Egor')
)

